### PR TITLE
Add Situation value object

### PR DESCRIPTION
**Pull request by tomphp**
# Motivation and Context
There is repeated logic throughout the services which verifies
the validity of the situation string for `ActionFeedback`
(limited to 100 chars). This class moves that validation
to a single place but `ActionFeedback` still expects a string.
For now the `toString()` method can be used but it would be
better if `ActionFeedback` accepted a proper `Situation` type
in the future (not sure if this is possible with the jaxb
generated class).

# What has changed
Add `Situation` value object.

# How to test?
This only adds to the API so just run the unit tests.

# Links
Used by https://github.com/ONSdigital/rm-notify-gateway/pull/32

### CODE REVIEW

File: src/main/java/uk/gov/ons/ctp/response/action/representation/exception/InvalidSituationException.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package uk.gov.ons.ctp.response.action.representation.exception;
+
+import uk.gov.ons.ctp.common.error.CTPException;
+import uk.gov.ons.ctp.response.action.representation.Situation;
+
+public class InvalidSituationException extends CTPException {
```
Comment (benjefferies) [inline src/main/java/uk/gov/ons/ctp/response/action/representation/exception/InvalidSituationException.java]: Should this be a checked exception, is it recoverable?

**Comment by tomphp** [comment]
[Create an auto-verifying notify endpoint service](https://trello.com/c/lZYrIXw7/118-create-an-auto-verifying-notify-endpoint-service)

File: src/main/java/uk/gov/ons/ctp/response/action/factory/ActionFeedbackFactory.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package uk.gov.ons.ctp.response.action.factory;
+
+import uk.gov.ons.ctp.response.action.message.feedback.ActionFeedback;
+import uk.gov.ons.ctp.response.action.message.feedback.Outcome;
+import uk.gov.ons.ctp.response.action.representation.Situation;
+
+public class ActionFeedbackFactory {
+  public static ActionFeedback create(String actionId, Situation situation, Outcome outcome) {
```
Comment (benjefferies) [inline src/main/java/uk/gov/ons/ctp/response/action/factory/ActionFeedbackFactory.java]: Feels like overkill to have a factory to create an instance of ActionFeedback 🤔. Why not give the constructor this signature

File: src/main/java/uk/gov/ons/ctp/response/action/representation/exception/InvalidSituationException.java (hunk 1 above)
Comment (tomphp) [reply src/main/java/uk/gov/ons/ctp/response/action/representation/exception/InvalidSituationException.java]: We were trying to follow what seemed to be convention in the code bases. As far as we can see, any occurrence of this exception would be a logic error (i.e. code mistake). Do you think we should change this to a `RuntimeException`?

File: src/main/java/uk/gov/ons/ctp/response/action/factory/ActionFeedbackFactory.java (hunk 2 above)
Comment (benjefferies) [reply src/main/java/uk/gov/ons/ctp/response/action/factory/ActionFeedbackFactory.java]: Oh that makes sense 👍. Forgot you mentioned that last week

File: src/main/java/uk/gov/ons/ctp/response/action/representation/exception/InvalidSituationException.java (hunk 1 above)
Comment (benjefferies) [reply src/main/java/uk/gov/ons/ctp/response/action/representation/exception/InvalidSituationException.java]: I reckon it's worth it being a RuntimeException. Saves the next person using it being confused

**Pull request merged**